src/Event/Productions/ProductionRepository.php

```diff
@@ -133,7 +133,7 @@ private function createSearchQuery(string $keyword): QueryBuilder
             ->from($this->getTableName()->toNative());
 
         if (!empty($keyword)) {
-            $query = $query->where('MATCH (name) AGAINST (:keyword)')
+            $query = $query->where('name LIKE CONCAT(\'%\', :keyword, \'%\')')
```

Comment:
I wonder if it also works like this
```
$query = $query->where('name LIKE \'%' .  :keyword . '\'%')
```

That would avoid a `CONCAT` to be called for every single row.

Reply:
Your proposal is not syntactically valid. MySQL always uses `CONCAT` for concatenation, no? Besides, why would `CONCAT` happen for every row but your syntax not?

Reply:
Indeed I made a mistake, it should have been
```
$query = $query->where('name LIKE \'%:keyword\'%');
```

MySQL does indeed use `CONCAT` to do the string concatenation but you can make use of DBAL here to make it a fixed string in which the `$keyword` variable is already replaced. So you get a where clause as `name LIKE '%movie%'`.

It's going to be called for every row because `CONCAT` is a MySQL function and the parameter you pass can be a column value from the current row. So it's default behaviour to execute it for every row even when the input parameter is a fixed string.

Reply:
In the meantime I switched back to `MATCH` based on @jdrieghe 's comment. https://github.com/cultuurnet/udb3-silex/pull/486/commits/f7522d3b068f4cdb1cd625ab2c0740bd517410f4

I tried to change it locally to `AGAINST(:keyword* IN BOOLEAN MODE)` but that results in:
```
An exception occurred while executing 'SELECT production_id, name, GROUP_CONCAT(event_id) as events FROM productions WHERE MATCH(name) AGAINST(?* IN BOOLEAN MODE) GROUP BY production_id' with params [\"mod\"]:\n\nSQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IN BOOLEAN MODE) GROUP BY production_id' at line 1
```

As I suspected and as is visible in the error message, DBAL doesn't replace the values itself but passes an actual prepared statement to MySQL as well as the bound values. It's MySQL who then does the replacement while executing the prepared statement. And MySQL doesn't understand `:keyword*`  because it only supports concatenation using `CONCAT`. But even if MySQL understood `:keyword*`, the concatenation would then still happen per row according to your info because it happens while executing the statement, not as some kind of `str_replace` in PHP before executing the statement.

Reply:
I was able to optimize it though with a tip from @jdrieghe to do the concatenation in PHP itself first: https://github.com/cultuurnet/udb3-silex/pull/486/commits/13081f80278e5e4cdf099ebaf0d6b0e1cfa9937d

Still not sure if it was really necessary. I'd assume MySQL would be smart enough by now to know that this particular CONCAT would always be the same for each row since it would only take a prepapered statement parameter and a hardcoded string, but with this solution we're 100% sure.